src/ray/gcs/gcs_server/gcs_monitor_server.cc

```diff
@@ -41,5 +46,81 @@ void GcsMonitorServer::HandleDrainAndKillNode(
   send_reply_callback(Status::OK(), nullptr, nullptr);
 }
 
+void GcsMonitorServer::PopulateNodeStatuses(rpc::GetSchedulingStatusReply *reply) const {
+  const auto &scheduling_nodes = cluster_resource_manager_.GetResourceView();
+  const auto &gcs_node_manager_nodes = gcs_node_manager_->GetAllAliveNodes();
+
+  for (const auto &pair : gcs_node_manager_nodes) {
+    const auto &node_id = pair.first;
+    const auto &gcs_node_info = pair.second;
+
+    const auto &it = scheduling_nodes.find(scheduling::NodeID(node_id.Binary()));
+    if (it == scheduling_nodes.end()) {
+      // A node may be in GcsNodeManager but not ClusterResourceManager due to
+      // a couple of edge cases/race conditions.
+      continue;
+    }
+    const auto &node_resources = it->second.GetLocalView();
+    const auto &available = node_resources.available.ToResourceMap();
```

Comment:
I think ToResourceMap should be a value right? So not necessary ref here?

Reply:
... how did this even compile...